docs/ios/open-source/manual-instrumentation/performance-monitoring.md

```diff
@@ -0,0 +1,363 @@
+    
+    // Payment logic
+    performPayment { success in
+        span?.setAttribute(key: "payment_success", value: success ? "true" : "false")
+        span?.end()
+        
+        completion(success)
+    }
+}
+
+func confirmOrder(completion: @escaping (Bool) -> Void) {
+    let span = Embrace.client?.startSpan(name: "order_confirmation")
+    
+    // Order confirmation logic
+    confirmOrderOnServer { success in
+        span?.setAttribute(key: "confirmation_success", value: success ? "true" : "false")
+        span?.end()
+        
+        completion(success)
+    }
+}
+```
+
+## Tracking UI Performance
+
+Monitor UI rendering and user interaction performance:
+
+```swift
+class ProductListViewController: UIViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        Embrace.client?.logMessage("ProductList viewDidLoad started", severity: .info)
+        
+        // Setup UI
+        setupUI()
+        
+        Embrace.client?.logMessage("ProductList viewDidLoad completed", severity: .info)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        let span = Embrace.client?.startSpan(name: "product_list_loading")
+        
+        // Load data
+        loadProducts { [weak self, weak span] products in
+            self?.updateUI(with: products)
+            
+            span?.setAttribute(key: "product_count", value: "\(products.count)")
+            span?.end()
+        }
+    }
+    
+    private func loadProducts(completion: @escaping ([Product]) -> Void) {
+        // Loading logic
+    }
+    
+    private func updateUI(with products: [Product]) {
+        let span = Embrace.client?.startSpan(name: "product_list_ui_update")
+        
+        // UI update logic
+        collectionView.reloadData()
+        
+        span?.end()
+    }
+}
+```
+
+## Custom Attributes for Performance Context
+
+Add custom attributes to provide context for performance analysis:
+
+```swift
+// Track device-specific attributes
+Embrace.client?.addSessionProperty(
+    name: "device_performance_tier",
+    value: determineDevicePerformanceTier() // e.g., "high", "medium", "low"
+)
+
+// Track user-specific attributes that might affect performance
+if let userSubscriptionTier = userManager.currentUser?.subscriptionTier {
+    Embrace.client?.addSessionProperty(name: "subscription_tier", value: userSubscriptionTier)
+}
+
+// Track app configuration that impacts performance
+Embrace.client?.addSessionProperty(name: "image_quality", value: appSettings.imageQuality.rawValue)
+Embrace.client?.addSessionProperty(name: "animations_enabled", value: appSettings.animationsEnabled ? "true" : "false")
+```
+
+## Performance Optimization Best Practices
+
+### Set Performance Budgets
+
+Define performance thresholds and log when they're exceeded:
+
+```swift
+func measureOperation(name: String, budget: TimeInterval, operation: () -> Void) {
```

Comment:
do we want to be doing this from the client? or is it better to measure it all and make sense of it on the dashboard